lib/teleterm/daemon/config.go

```diff
@@ -25,10 +27,28 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type (
+	cluster interface {
+		GetServers(ctx context.Context) ([]clusters.Server, error)
+	}
+	storage interface {
+		GetByURI(uri string) (cluster, error)
+	}
+)
+
+// This is used to wrap a real clusters.Storage so it meets our interface
+type StorageWrapper struct {
```

Comment:
I would ditch the wrapper - code usage is king here, and should be enough to guarantee conformance to the interfaces. If there are no uses, then a temporary `var _ storage = &MyType{}` is enough to assert conformance.

Your method signatures are _not_ an accident, they are purposeful.